### PR TITLE
Clean up old Proto code

### PR DESCRIPTION
**Pull request by KnathanM**
## Description
#536 changed many protocol class uses. It looks like some parts of the code were missed. 

Also at some point `chemprop/v2/models/hparams.py` was renamed to both `chemprop/v2/utils/hparams.py` and `chemprop/v2/nn/hparams.py`. It looks like the intention was for it to be in `nn` so I have removed the duplicate in `utils`.

## Questions
The changes in `chemprop/cli/utils/parsing.py` fix an error that prevented the cli from running. It would be good to include the CLI in our auto checks to catch these errors earlier. How do we want to add that?

### CODE REVIEW

**Review by davidegraff — APPROVED** [review]
👍

**Comment by davidegraff** [comment]
We could write another integration test that mimics running from the CLI via [`subprocess.run`](https://docs.python.org/3/library/subprocess.html#subprocess.run). Ideally it should mimic the form of the package-level integration tests.

Ex:
1. using the same underlying fixture data, write a temporary file to mimic an input CSV
2. construct the argument vector using `@pytest.fixture`s to test test a variety of model arch's
3. parse the output file for an overall training loss (i.e., loss of the final model on the training dataset)
4. ensure that the training loss is $\leq \epsilon$ for some small $\epsilon$
